airbyte-cdk/python/airbyte_cdk/test/http/response_builder.py

```diff
@@ -0,0 +1,131 @@
+
+    def _validate_template(self) -> None:
+        paths_to_validate = [
+            ("_id_path", self._id_path),
+            ("_cursor_path", self._cursor_path),
+        ]
+        for field_name, field_path in paths_to_validate:
+            try:
+                self._validate_field(field_name, field_path)
+            except KeyError:
+                raise ValueError(f"{field_name} `{field_path}` was provided but it is not part of the template")
+
+    def _validate_field(self, field_name: str, field_path: Optional[List[str]]) -> None:
+        if field_path and not _extract(field_path, self._record):
+            raise ValueError(f"{field_name} `{field_path}` was provided but it is not part of the template")
+
+    def with_id(self, identifier: Any) -> "RecordBuilder":
+        self._set_field("id", self._id_path, identifier)
+        return self
+
+    def with_cursor(self, cursor_value: Any) -> "RecordBuilder":
+        self._set_field("cursor", self._cursor_path, cursor_value)
+        return self
+
+    def _set_field(self, field_name: str, path: Optional[List[str]], value: Any) -> None:
+        if not path:
+            raise ValueError(
+                f"{field_name}_path was not provided and hence, the record {field_name} can't be modified. Please provide `id_field` while "
+                f"instantiating RecordBuilder to leverage this capability"
+            )
+        _replace_value(self._record, path, value)
+
+    def build(self) -> Dict[str, Any]:
+        return self._record
+
+
+class HttpResponseBuilder:
+    def __init__(self, template: Dict[str, Any], records_path: List[str], pagination_strategy: Optional[Callable[[Dict[str, Any]], None]]):
+        self._response = template
+        self._records: List[RecordBuilder] = []
+        self._records_path = records_path
+        self._pagination_strategy = pagination_strategy
+        self._status_code = 200
+
+    def with_record(self, record: RecordBuilder) -> "HttpResponseBuilder":
+        self._records.append(record)
+        return self
+
+    def with_pagination(self) -> "HttpResponseBuilder":
+        if not self._pagination_strategy:
+            raise ValueError(
+                "`pagination_strategy` was not provided and hence, fields related to the pagination can't be modified. Please provide "
+                "`pagination_strategy` while instantiating ResponseBuilder to leverage this capability"
+            )
+        self._pagination_strategy(self._response)
+        return self
+
+    def with_status_code(self, status_code: int) -> "HttpResponseBuilder":
+        self._status_code = status_code
+        return self
+
+    def build(self) -> HttpResponse:
+        _replace_value(self._response, self._records_path, [record.build() for record in self._records])
+        return HttpResponse(json.dumps(self._response), self._status_code)
+
+
+def _get_unit_test_folder() -> Path:
+    path = Path(os.getcwd())
+    while path.name != "unit_tests":
```

Comment:
It seems like this is a bad assumption as tests are failing in the pipeline as shown [here](https://github.com/airbytehq/airbyte/actions/runs/7049202790/job/19187228648?pr=32983#step:8:24979). I'll try to figure out how we can identify the resource file without relying on the `unit_tests` folder

Reply:
I added the ability to pass the path where to look for the `unit_tests` directory. The user can pass `__file__` from the test which should be pretty easy